nubis/puppet/files/nubis-secret

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# /usr/local/bin isn't set yet in our PATH
+export PATH=/usr/local/bin:$PATH
+
+USER_DATA=$(curl -fqs http://169.254.169.254/latest/user-data)
+eval "$USER_DATA"
+
+REGION=$(curl -s http://169.254.169.254/latest/dynamic/instance-identity/document | jq '.region' -r)
+
+CREDSTASH="/usr/local/bin/credstash"
+
+ACTION=$1
+shift
+
+SECRET=$1
+shift
+
+if [ "$ACTION" != "get" ] || [ ! "$SECRET" ]; then
+    echo "Usage: $0 get secret"
+    exit
+fi
+
+$CREDSTASH --region "$REGION" "$ACTION" "$SECRET" "environment=$NUBIS_ENVIRONMENT" "region=$REGION" "service=$NUBIS_PROJECT"
```

Comment:
You might want to validate the assumptions you are making around existing variables like $NUBIS_ENVIRONMENT ans $NUBIS_PROJECT. At least test for their existence and throw a valid error. Additionally you may want to use logger or a similar method to uplift these errors to the logging stream to aid in troubleshooting in the future.

I would consider the validation portion necessary and the logging part useful but not mandatory.